app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -314,23 +315,24 @@ open class FenixApplication : LocaleAwareApplication() {
 
     private fun enableStrictMode() {
         if (Config.channel.isDebug) {
-            StrictMode.setThreadPolicy(
-                StrictMode.ThreadPolicy.Builder()
-                    .detectAll()
+            Handler().postAtFrontOfQueue(Runnable {
+                StrictMode.setThreadPolicy(
+                    StrictMode.ThreadPolicy.Builder()
+                        .detectAll().penaltyLog()
```

Comment:
I feel as if we should set `penaltyDialog()` instead of `penaltyLog()` and (maybe, not sure about this one, what do you think about this @mcomella) using `penaltyDeathOnNetwork()` since it would allow us to kill the app on network operations on the MT on start up.

Reply:
Yup done! I am still waiting on `penaltyDeathOnNetwork()` suggestion from @mcomella

Reply:
MarcLeclair connected with me async and I said something like,

> Hmm, that's curious – I thought Android automatically killed you, even in production, on main thread network access (but I guess MLS bug proves that wrong)
>
> I'm down but maybe land it in a separate patch so it's easy to see if it causes problems?

Reply:
So yes, penaltyDeathOnNetwork sounds good to me.